federation/pkg/kubefed/init/init.go

```diff
@@ -582,6 +601,42 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 	return clientset.Extensions().Deployments(namespace).Create(dep)
 }
 
+func waitForPods(clientset *client.Clientset, name, nameSpace string) error {
+	err := wait.PollInfinite(podWaitInterval, func() (bool, error) {
+		podList, err := clientset.Core().Pods(nameSpace).List(api.ListOptions{})
+		if err != nil {
+			return false, nil
+		}
+		for _, pod := range podList.Items {
+			if strings.Contains(pod.Name, name) && pod.Status.Phase == "Running" {
```

Comment:
I have been thinking quite a bit about this. I think the right thing to do here is to get the deployment status, then get the pod by using its selector and then check the pod's phase. It's Ok to do that in a follow up PR. But at least do a `strings.HasPrefix()` check instead of a `strings.Contains()` check here.